srk/src/polynomialLattice.ml

```diff
@@ -1,53 +1,60 @@
 open Polynomial
+open BatPervasives
 
 module L = Log.Make(struct let name = "srk.polyLattice" end)
 
+type context =
+  { ctx : LinearQQXs.context
+  ; monomials : Monomial.t list
+  }
+
 (** Affine lattice is reduced with respect to ideal.
     The lattice is unchanged whether the affine lattice is reduced or not.
  *)
 type t =
   { ideal : Rewrite.t
   ; affine_lattice : IntLattice.t
-  ; affine_context : LinearQQXs.context
+  ; affine_context : context
   }
 
-(* Some graded order *)
+(* Use [degrevlex] order so that the intersection of the affine lattice with
+   QQ[Y] can be computed as the intersection of the basis with QQ[Y].
+ *)
 let monomial_order = Polynomial.Monomial.degrevlex
 
 module MonomialSet =
   BatSet.Make(struct type t = Monomial.t
-                     let compare x y = match Monomial.degrevlex x y with
+                     let compare x y = match monomial_order x y with
                        | `Lt -> -1
                        | `Eq -> 0
                        | `Gt -> 1
               end)
 
-(* TODO: Check if we can disregard monomial order and just use [min_context].
-   In projection, we need monomials weighted with fresh dimensions/variables
-   to occur as higher dimensions when densifying into vectors for integer
-   lattice projection.
- *)
 let make_context polys =
-  let monos = List.fold_left
-                (fun monos poly ->
-                  BatEnum.fold (fun s (_coeff, mono) -> MonomialSet.add mono s)
-                    monos (QQXs.enum poly))
-                MonomialSet.empty polys
+  let monos = polys
+              |> List.fold_left
+                   (fun monos poly ->
+                     QQXs.enum poly
+                     |> BatEnum.fold (fun s (_coeff, mono) -> MonomialSet.add mono s)
+                          monos)
+                   MonomialSet.empty
+              |> MonomialSet.elements
   in
-  LinearQQXs.make_context (MonomialSet.elements monos)
-
-(*
+  { ctx = LinearQQXs.make_context monos
+  ; monomials = monos
+  }  
+  
 let zero =
```

Comment:
Is this empty or {0}?

Reply:
{0}; I believe Ideal.make [] is {0}? (Also, dropping zero since it's really not used anywhere.)